Data/Functor/Foldable/TH.hs

```diff
@@ -0,0 +1,141 @@
+{-# LANGUAGE TemplateHaskell #-}
+module Data.Functor.Foldable.TH
+  ( makeBaseFunctor
+  ) where
+
+import Data.Bifunctor (first)
+import Data.Functor.Foldable
+import Language.Haskell.TH
+
+makeBaseFunctor :: Name -> DecsQ
```

Comment:
> I'm not sure how much different types this should support, e.g. how to generate names for infix constructors (what e.g. `lens` does then?).

IIRC, I think `lens` just chokes on infix constructors. :)

But if you _do_ want to support infix entities, the `singletons` library has a nice, uniform [(re)naming convention](https://github.com/goldfirere/singletons/tree/dd8057c973224dbaf58bd4dbcec9eb0e54c9ae20#on-names) that could be easy to adapt. Alternatively, you could just document the fact that infix things aren't supported yet.

> I guess supporting `GADTs` isn't good ROI, at least for now.

Don't interpret that sentence as me _requesting_ that you implement support for any of these exotic GHC features—I just request that we make it clear in the documentation that only Haskell98 datatypes have been tested.